dask_expr/collection.py

```diff
@@ -707,6 +709,21 @@ def select_dtypes(self, include=None, exclude=None):
     def eval(self, expr, **kwargs):
         return new_collection(Eval(self.expr, _expr=expr, expr_kwargs=kwargs))
 
+    def set_index(self, other, drop=True, sorted=False, divisions=None):
+        if isinstance(other, Series):
+            if other._name == self.index._name:
+                return self
+        else:
+            if other == self.index.name:
+                return self
```

Comment:
```suggestion
        elif other == self.index.name:
            return self
```

Not a big deal, but maybe simpler